lib/collection/src/operations/conversions.rs

```diff
@@ -499,7 +500,9 @@ impl From<api::grpc::qdrant::OptimizersConfigDiff> for OptimizersConfig {
             flush_interval_sec: optimizer_config.flush_interval_sec.unwrap_or_default(),
             max_optimization_threads: optimizer_config
                 .max_optimization_threads
-                .map(|n| n as usize),
+                .map_or(MaxOptimizationThreads::Auto, |n| {
+                    MaxOptimizationThreads::Threads(n as usize)
+                }),
```

Comment:
This means that even if `max_optimization_threads` is not specified in a grpc client, it will set it to `Auto`, which is not necessarily the intention.

Reply:
Thanks for the review and detailed comments. I will take another pass at it. 

My initial intention was to not make any change in the grpc client so as to avoid any need of deprecation (tags) and worry about backward compatibility. Hence I decided to only change the inners i.e the ```OptimizersConfig``` and handle remaps via ```conversions``` --  but it seems broken. I will rerun integration tests locally to see what is broken or follow what you have suggested.

Reply:
Unfortunately we don't have a test which would break with this change yet, nor an easy framework for testing grpc interface. Most grpc changes are just validated with the compiler.

For REST, though, a test could be made to validate that setting a number or `"auto"` works ok.